docs/api/navigators/StackNavigator.md

```diff
@@ -118,7 +118,7 @@ All `navigationOptions` for the `StackNavigator`:
 
 - `title` - a title (string) of the scene
 - `header` - a config object for the header bar:
-  - `visible` - Boolean toggle of header visibility. Only works when `headerMode` is `screen`.
+  - `bar` - React Element to display as a header or `null` to hide it. Only works when `headerMode` is `screen`
```

Comment:
`header.bar` and `headerComponent` are redundant at global level. I think there's no reason to have both.

The `navigation` that gets passed is only for current screen. You don't even know if there's something before this screen in the stack to decide if you want to render a back button or not.